### PR TITLE
[WIP] Add featured items

### PR DESCRIPTION
**Pull request by callumlocke**

### CODE REVIEW

File: components/featured-items/_main.scss [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+			display: block;
+			position: absolute;
+			top: 0;
+			left: 0;
+		}
+	}
+}
+
+.featured-items__headline {
+	font: 400 22px/1.2 oFontsGetFontFamilyWithFallbacks(FinancierDisplayWeb);
```
Comment (andygnewman) [inline components/featured-items/_main.scss]: We generally use o-typography for fonts https://github.com/Financial-Times/o-typography

File: server/lib/inject-featured-items.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+ * What this does:
+ *
+ * - Downloads a list from the CAPI, plus content for items in that list.
+ * - Injects multiple rows of articles (depending on the layoutHint, controlled
+ *   from Methode) into the `items` array.
+ *
+ * Example list: http://api.ft.com/lists/7a084d44-b95a-11e5-b151-8e15c9a029fb
+ */
+
+import logger from 'ft-next-logger';
```
Comment (andygnewman) [inline server/lib/inject-featured-items.js]: Will be upgrading to a more recent logger which will need this changed to import {logger} https://github.com/Financial-Times/next-stream-page/pull/872

File: server/lib/inject-featured-items.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	{
+		const layoutDescriptor = layouts[list.layoutHint] || layouts['standard'];
+
+		let i = 0;
+		for (const [indexInStream, numItems] of layoutDescriptor.split(',').map(row => row.trim().split(':'))) {
+			const numRemaining = featuredItems.length;
+			if (!numRemaining) break;
+
+			items.splice(Number(indexInStream) + i, 0, {
+				id: `${listId}_${indexInStream}`, // react needs a unique id. not sure if this needs to be something nicer for tracking.
```
Comment (andygnewman) [inline server/lib/inject-featured-items.js]: This is how we've done react keys for the stream items https://github.com/Financial-Times/next-stream-page/blob/master/components/stream/index.js#L178

File: components/featured-items/_main.scss [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+	border-bottom: 1px dotted #333333;
+}
+
+.featured-items__subheading {
+	color: #646261;
+	font: italic 200 15px/1.2 oFontsGetFontFamilyWithFallbacks(FinancierDisplayWeb);
+	margin: 0 0 0.5em;
+}
+
+.featured-items__end-matter> .o-date {
```
Comment (andygnewman) [inline components/featured-items/_main.scss]: we have a common styling class for time date .c-time-date that you could use

**Comment by callumlocke** [comment]
closing in favour of https://github.com/Financial-Times/next-stream-page/pull/885

**Comment by callumlocke** [comment]
still using this branch for reference, will delete it soon

**Pull request closed without merging**